tests/protocol/test_wire.py

```diff
@@ -442,6 +443,29 @@ def test_report_vm_status(self, *args):
         }
         self.assertEqual(json.dumps(v1_vm_status), actual.to_json())
 
+    @patch("azurelinuxagent.common.utils.restutil.http_request")
+    def test_send_event(self, mock_http_request, *args):
+        mock_http_request.return_value = MockResponse("", 200)
+
+        event_str = u'ćevapčići'
+        client = WireProtocol(wireserver_url).client
+        client.send_event("foo", event_str)
+
+        first_call = mock_http_request.call_args_list[0]
+        args, kwargs = first_call
+        method, url, data_received = args
+        headers = kwargs['headers']
+
+        data_format = ('<?xml version="1.0"?>'
+                       '<TelemetryData version="1.0">'
+                       '<Provider id="{0}">{1}'
+                       '</Provider>'
+                       '</TelemetryData>')
+        data_encoded = data_format.format("foo", event_str.encode("utf-8"))
```

Comment:
I think the test has too much knowledge of the internal implementation of send_event (it needs to know the exact format of the xml), which makes it fragile in the sense that if the implementation changes the test may break. Maybe using a longer body (a longer test string) and contains() instead of strict equality is a better option?

Reply:
Agreed, changed test to be more robust.